### PR TITLE
[17.0][MIG] web_favicon: Migration to 17.0 

### PR DESCRIPTION
**Pull request by dc-oerp**
preserve commit history from V12 to V17

Previously in this PR:
https://github.com/OCA/web/pull/2765#issuecomment-1989168258

### CODE REVIEW

**Comment by dc-oerp** [comment]
Hi @pedrobaeza 

I have closed the other PR and created a new PR here, Please let me know if there is still any issue with this migration.

Thank you

**Comment by pedrobaeza** [comment]
Please check CI

**Comment by dc-oerp** [comment]
All fixed

**Review by ioans73 — APPROVED** [review]
Functional review

File: web_favicon/views/res_company_view.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                    name="favicon"
+                    widget="image"
+                    class="float-start oe_avatar"
+                    groups="base.group_no_one"
+                />
+            </xpath>
+        </field>
+    </record>
+
+    <template id="favicon_icon" inherit_id="web.layout" name="Web layout">
```
Comment (pedrobaeza) [inline web_favicon/views/res_company_view.xml]: This shouldn't go here, but on `templates.xml` file.

File: web_favicon/views/res_company_view.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 OERP Canada <https://www.oerp.ca>
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). -->
```
Comment (pedrobaeza) [inline web_favicon/views/res_company_view.xml]: This is AGPL, not LGPL. And please call the file `res_company_views.xml`.

File: web_favicon/tests/__init__.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# Copyright 2024 OERP Canada <https://www.oerp.ca>
```
Comment (pedrobaeza) [inline web_favicon/tests/__init__.py]: Copyright is not put in init files.

File: web_favicon/tests/__init__.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
```
Comment (pedrobaeza) [inline web_favicon/tests/__init__.py]: LGPL > AGPL

**Review by pedrobaeza — COMMENTED** [review]
Tested on runboat and working. Just few comments.

**Comment by dc-oerp** [comment]
Hi @pedrobaeza 

Thanks for the comments.
I faced one issue where after I fixed the comments, and ran pre-commit locally, it passed without any issues.
However, after I pushed the code, it showed that there was an issue with Pylint.

Could you please take a look?

Thanks

File: web_favicon/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+# Copyright 2016 Pedro M. Baeza
+# Copyright 2024 OERP Canada <https://www.oerp.ca>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Custom shortcut icon",
+    "version": "17.0.1.0.0",
+    "author": "Therp BV, "
+    "Tecnativa, "
+    "OERP Canada"
```
Comment (pedrobaeza) [inline web_favicon/__manifest__.py]:
```suggestion
    "OERP Canada,"
```

File: web_favicon/static/description/icon.png
Comment (pedrobaeza) [inline web_favicon/static/description/icon.png:1]: Please keep the previous image

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**